core/src/types/operator/operator.rs

```diff
@@ -158,16 +158,36 @@ impl Operator {
         }
     }
 
-    /// Get current path's metadata.
+    /// Get given path's metadata.
     ///
     /// # Notes
     ///
     /// For fetch metadata of entries returned by [`Lister`], it's better to use [`list_with`] and
     /// [`lister_with`] with `metakey` query like `Metakey::ContentLength | Metakey::LastModified`
-    /// so that we can avoid extra requests.
+    /// so that we can avoid extra stat requests.
+    ///
+    /// # Behavior
+    ///
+    /// `test` and `test/` may vary in some services such as S3. However, on a local file system,
+    /// they're identical. Therefore, the behavior of `stat("test")` and `stat("test/")` might differ
+    /// in certain edge cases. Always use `stat("test/")` when you need to access a directory if possible.
+    ///
```

Comment:
The essence may still need to be noted here. Those services rely on prefixes.Or like WebDAV, which locates resources by their url, there is no way to determine if it is a folder or not before accessing them. That's why this happens.

Reply:
Yes, there are services that don't support `create_dir`, `stat` just treat them as simple file. I will add comments in following PRs.